### ISSUE TITLE
onClose not called when provided to Micromodal.init

### ISSUE THREAD

**Issue by CoreyBurkhart** [ghosh/Micromodal#354]
**Repo**
```js
MicroModal.init({
  onClose() { console.log('close') }
})

MicroModal.show('my-modal')
MicroModal.close('my-modal')
```
**Expectation:** "close" will be logged to the console
**Actual:** Nothing is logged to the console

When I provide the config to `show`, `onClose` gets called properly
```js
MicroModal.show('my-modal', {
  onClose() { console.log('close') }
})
MicroModal.close('my-modal') // logs "close"
```

**Comment by mrdarkside** [comment]
```
MicroModal.init({
  onClose: () => { console.log('close') }
})
```
no?

**Comment by mattgaspar** [comment]
Calling `MicroModal.show('my-modal')` seems to clear any configuration set from `MicroModal.init({ options... })`
A work around that seems to work is to not use `init` if you need to use `show` but then you can't use `data-micromodal-trigger` so you need to wire up your own event handlers.  Or pass in the same options for `init` and `show`!

**Comment by edadams** [comment]
@mattgaspar 's second workaround worked for me! I was doing `init` with a bunch of options I needed and then manually displaying my modal via `show`. It's really quite baffling that calling `show` or `hide` would nuke the config options.

It'd be _vastly_ more intuitive and easy to work with this library if the configuration options were persistent and `show` or `close` didn't delete them.

**Comment by OliverLSanz** [comment]
> It'd be vastly more intuitive and easy to work with this library if the configuration options were persistent and show or close didn't delete them.

I agree. Global configuration could be overridden with configuration passed to `show`. But if nothing is passed to `show`  global configuration should be used instead.

**Comment by mestrini** [comment]
In case someone still needs to keep plugin's default configuration values like `openTrigger: 'data-micromodal-trigger' ` or the ones set at `init()` I've implemented a way to not have them overriden every time a `show()` is called.

Since I don't have _npm_ system I got the source from https://github.com/ghosh/Micromodal/blob/master/lib/src/index.js and worked on it

1. append the following line after the plugin options are assigned at `init()`
```MicroModal.defaults = options;```

```
const init = config => {
        // Create an config object with default openTrigger
        const options = Object.assign({}, { openTrigger: 'data-micromodal-trigger' }, config)
        MicroModal.defaults = options; //added by mestrini
```
2. edit the line where config options are passed to the `show()` call to look like the one with the comment **_edited by mestrini_**
```
const show = (targetModal, config) => {
    const options = config === undefined ? MicroModal.defaults : extendDefaults(config); //edited by mestrini
    
    options.targetModal = targetModal
```
3.
add following function copied from https://gist.github.com/cferdinandi/4f8a0e17921c5b46e6c4 before the line 
    `return { init, show, close }`
```
const extendDefaults = (options) => { //added by mestrini
        let defaults = MicroModal.defaults;
        let extended = {};
        let prop;

        for (prop in defaults) {
            if (Object.prototype.hasOwnProperty.call(defaults, prop)) {
                extended[prop] = defaults[prop];
            }
        }
        for (prop in options) {
            if (Object.prototype.hasOwnProperty.call(options, prop)) {
                extended[prop] = options[prop];
            }
        }
        return extended;
    }
```

That's it. Hope it's useful. 

Comments and corrections are welcomed. Cheers

**Comment by dkniffin** [comment]
I've confirmed this bug. If anyone wants to create a PR, that'd be helpful. Otherwise, I'll get to this when I can.

**Comment by dkniffin** [comment]
This should now be resolved in 0.7.0. Let me know if the issue continues.
